src/Prototype/Types/Secret.hs

```diff
@@ -26,6 +26,8 @@ module Prototype.Types.Secret
   ( Secret(..)
   , SecretExp(..)
   , exposeSecret
+  -- * Compare secrets
```

Comment:
- Nice module :)
- But really, do you plan do compare a lot of secrets ?
- The thing that could be useful here would be a constant-time comparison.

Reply:
Thanks; the idea is to have users authenticated and even their login "tokens" be compared on each protected route authorization. And we can also then derive a `Show` on all types containing secrets and rest assured that the secrets will be shown in the appropriate form (eg. masked or not); and not have to bother about that, or end up with human introduced errors in which things like passwords etc. get logged.